src/logic/tokens/store/actions/fetchTokens.js

```diff
@@ -50,6 +50,27 @@ export const containsMethodByHash = async (contractAddress: string, methodHash:
   return byteCode.indexOf(methodHash.replace('0x', '')) !== -1
 }
 
+let tokensInstanceCache = {}
```

Comment:
we don't need a different cache for token instances, what we should do is:
1. Persist token list stored in redux also in localStorage
2. Whenever we fetch transactions, get token info from redux store instead of doing the request
3. If there's no such token, fetch its info and put into localstorage
4. Also fetch tokens tokens on app startup once and somehow get notified when token list on backend updates

Reply:
Hi @Uxio0 would you be able to imlement etags also for the fetch tokens method? so I can be notified when the list updates

Reply:
If you are talking about the list on the relay service, you can just use `count` to check if the list has been modified, but that list is not supposed to change very often, how frequently are you updating it?

Reply:
@mikheevm 
> 1. Persist token list stored in redux also in localStorage

 Do you think is really needed to persist the list in the localStorage? I think requesting it when the app load won't hurt the performance.  Using Redux instead of a new cache, makes sense to me.

Reply:
@nicosampler performance won't be hurt, but data usage would be :)